ci/scripts/lib/bazel_query_test.py

```diff
@@ -0,0 +1,147 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import re
+import unittest
+from unittest.mock import Mock
+
+from bazel_query import BazelQuery, BazelQueryRunner
+
+
+class TestBazelQuery(unittest.TestCase):
+
+    def test_rule_exact(self):
+        query = BazelQuery.rule_exact("foo", "bar")
+        self.assertEqual(query, 'kind("^foo rule$", bar)')
+
+    def test_tag_exact(self):
+        query = BazelQuery.tag_exact("manual", "foo")
+        self.assertEqual(query, 'attr("tags", "[\\[ ]manual[,\\]]", foo)')
+
+    def test_regex_for_tag(self):
+        regex = BazelQuery.regex_for_tag("foo")
+        self.assertEqual(regex, '[\\[ ]foo[,\\]]')
+
+        # Regex doesn't match lists without "foo".
+        self.assertFalse(re.search(regex, "[]"))
+        self.assertFalse(re.search(regex, "[bar]"))
+        self.assertFalse(re.search(regex, "[bar, baz]"))
+
+        # Regex does not match superstrings of "foo".
+        self.assertFalse(re.search(regex, "[foobar]"))
+        self.assertFalse(re.search(regex, "[barfoo]"))
+        self.assertFalse(re.search(regex, "[foofoo]"))
+
+        # Regex matches "foo" in any position.
+        self.assertTrue(re.search(regex, "[bar, foo, baz]"))
+        self.assertTrue(re.search(regex, "[bar, foo]"))
+        self.assertTrue(re.search(regex, "[foo, baz]"))
+        self.assertTrue(re.search(regex, "[foo]"))
+
+
+class TestFindTargetsWithBannedChars(unittest.TestCase):
+
+    def test_no_test_suites(self):
+        backend = Mock()
+        backend.return_value = []
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [])
+
+    def test_one_target_empty_string(self):
+        """Bazel does not allow targets to have empty names.
+
+        This test simply documents how we respond to the impossible scenario.
+        """
+        backend = Mock()
+        backend.return_value = [""]
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [])
+
+    def test_only_good_chars(self):
+        backend = Mock()
+        backend.return_value = ["//foo:bar", "//bar_baz:foo"]
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [])
+
+    def test_only_bad_chars(self):
+        backend = Mock()
+        backend.return_value = ["!@#$", "^&*()", "\x01"]
+        bazel = BazelQueryRunner(backend=backend)
+        targets = bazel.find_targets_with_banned_chars()
+        self.assertEqual(list(targets), [
```

Comment:
Maybe: this shouldn't enforce order, but it's probably fine

Reply:
Apparently `assertCountEqual` is a thing. Might as well use that :)